plugins/org.eclipse.n4js.transpiler.es/src/org/eclipse/n4js/transpiler/es/transform/RestParameterTransformation.xtend

```diff
@@ -13,72 +13,79 @@ package org.eclipse.n4js.transpiler.es.transform
 import org.eclipse.n4js.n4JS.FormalParameter
 import org.eclipse.n4js.n4JS.FunctionDefinition
 import org.eclipse.n4js.transpiler.Transformation
+import org.eclipse.n4js.transpiler.TransformationDependency.Optional
 import org.eclipse.n4js.transpiler.TransformationDependency.RequiresBefore
 import org.eclipse.xtext.EcoreUtil2
 
 import static org.eclipse.n4js.transpiler.TranspilerBuilderBlocks.*
 
 /**
+ * Transforms ES2015 rest parameters to an ES5 equivalent.
  *
  * Dependencies:
  * <ul>
+ * <li>requiresBefore {@link ArrowFunction_Part1_Transformation}:<br>
+ *     the transpiled output code for varargs requires the implicit local arguments parameter, which is not available in
+ *     arrow functions; therefore, this transformation relies on arrow functions already having been transformed into
+ *     ordinary function expressions.
  * <li>requiresBefore {@link BlockTransformation}:<br>
  *     the block transformation sometimes wraps the entire body into a newly created function (for async functions);
  *     the handling of varargs added below must NOT be wrapped in such an inner function; the easiest way to ensure this
  *     is to execute BlockTransformation before this transformation.
  * </ul>
  */
-@RequiresBefore(BlockTransformation)
-class FormalParameterTransformation extends Transformation {
+@Optional(ES5plus)
+@RequiresBefore(ArrowFunction_Part1_Transformation, BlockTransformation)
```

Comment:
"@RequiresBefore(A)" means: this transformation requires annotation A to appear before itself (see API doc).

see prev. response above